content/jobs.md

```diff
@@ -38,41 +38,190 @@ Schema:
+    * **name** [String, required]: Name of the link to find.
+    * **type** [String, required]: Type of the link to be found. This is an
+      arbitrary naming. Usual and conventional types are `address` when the
+      link goal is to expose a Bosh DNS name atht allows accessing the
+      instances of the group. Usually typed by technology, like `mysql`,
+      `postgres`, `cassandra`, etc. Anything that makes sense is relevant and
+      matters.
+    * **optional** [Boolean, optional]: Whether finding an matching link is
+      optional (when `true`) or mandatory (when `false`. Default is `false`,
+      so optinal links must be explicitly declared as such.
+* **provides** [Array, optional]: Links that are exposed to other jobs for
+  rendering their ERB templates.
+    * **name** [String, required]: Name of the exposed link.
+    * **type** [String, required]: Type of the exposed link.
+    * **properties** [Array, optional]: List of property keys in dot notation
+      (same as **properties.&lt;name>** below)
 * **properties** [Hash, optional]: Configuration options supported by the job.
-    * **\<name\>** [String, required]: Property key in dot notation. Typical properties include account names, passwords, shared secrets, hostnames, IP addresses, port numbers, and descriptions.
-        * **description** [String, required]: Describes purpose of the property.
-        * **example** [Any, optional]: Example value. Default is `nil`.
-        * **default** [Any, optional]: Default value. Default is `nil`.
+    * **&lt;name>** [String, required]: Property key in dot notation. Typical
+      properties include account names, passwords, shared secrets, hostnames,
+      IP addresses, port numbers, and descriptions.
+        * **description** [String, required]: Describes purpose of the
+          property. This is not used by the Director, but is displayed in job
+          configuration details provided by the [release index](/releases).
+        * **type** [String, optional]: The type of the property. This is only
+          a convention for release authors to provide a type when they
+          estimate it useful. Example: `type: certificate`.
+        * **example** [Any, optional]: Example value, to be displayed in the
+          [release index](/releases). Default is `nil`.
+        * **default** [Any, optional]: The default value for the property.
+          Default is `nil`.
+
+!!! Note
+    Within a peoperty definition, `default` is used by the Director, and
+    `description`, `default` and `example` are displayed by the
+    [release index](/releases). In turns, other keys like `type` are used only
+    for convenience, like Concourse does `env` keys in the
+    [“web” job definition][concourse_web_spec]. Indeed, the schema is not
+    formally validated by the Director when registering a release job.
+
+[concourse_web_spec]: https://github.com/concourse/concourse-bosh-release/blob/8d2cfa0/jobs/web/spec#L68-L71
```

Comment:
In terms of providing examples, would it be more appropriate to use an example from the larger `cloudfoundry` ecosystem, rather than something outside of it?

Reply:
Well, in terms of Bosh Release authoring practice, I’ve noticed some years ago that the Concourse team was a bit ahead, even compared to other Pivotal teams. More than others, they were trying to question how to do things right, and be more expressive. Thus I’ve often designated the Concourse Bosh Release as a reference in terms of release authoring.

There are also the bosh releases that I’ve work on, which demonstrate the adoption of the `type` property for more clarity in job’s `spec` files:
- https://github.com/gstackio/traefik-boshrelease/blob/b375aec/jobs/traefik/spec#L131-L134
- https://github.com/gstackio/gk-kong-boshrelease/blob/89e02c5/jobs/kong/spec#L89-L93
- https://github.com/gstackio/gk-consul-boshrelease/blob/master/jobs/consul/spec#L151-L159
- https://github.com/orange-cloudfoundry/cassandra-boshrelease/blob/48248cf/jobs/cassandra/spec#L681-L697